### PR TITLE
Completed PreCourse-1. Please Review

### PR DESCRIPTION
**Pull request by owaiskazi19**

### CODE REVIEW

File: Exercise_1.py [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
      def isEmpty(self):
+          if len(self.res):
+               return True
          
      def push(self, item):
+          self.res.append(item)
          
      def pop(self):
-        
+        self.res.pop()
```
Comment (samarthjj) [inline Exercise_1.py:13]: Please add edge case to prevent from operating on a empty stack.

File: Exercise_4.py [hunk 2]
```diff
@@ -7,15 +7,50 @@ def __init__(self, data):
...
+    # Recursive Solution
+    # if temp: 
+    #     inorder(temp.left)
+    #     print (temp.key, end = " ")
+    #     inorder(temp.right)
+
+    # Iterative solution
+    stack = []
+    curr = temp
+    while True:
```
Comment (samarthjj) [inline Exercise_4.py:20]: Good work on implementing the iterative solution

**Comment by samarthjj** [comment]
Owais, Good work on completing Pre_Course-1. Please review the inline comments.

**Comment by samarthjj** [comment]
Owais, Good work on completing Pre_Course-1. Please review the inline comments.

File: Exercise_1.py (hunk 1 above)
Comment (owaiskazi19) [reply Exercise_1.py]: Done!

File: Exercise_4.py (hunk 2 above)
Comment (owaiskazi19) [reply Exercise_4.py:20]: Thanks!

**Pull request closed without merging**